yaml_edit/lib/src/editor.dart

```diff
@@ -417,17 +407,15 @@ class YamlEditor {
   /// - 2 # comment 2
   /// '''
   /// ```
-  YamlNode remove(Iterable<Object> path) {
-    ArgumentError.checkNotNull(path, 'path');
-
-    SourceEdit edit;
-    YamlNode expectedNode;
+  YamlNode remove(Iterable<Object?> path) {
+    SourceEdit? edit;
```

Comment:
How about:
```suggestion
    late SourceEdit edit;
```

All the if-branches below will assign a value to `edit`. So we will write to it before we read from it, no matter what happens.